### DISCUSSION TITLE
batch number for paths.types

### DISCUSSION THREAD

**Discussion by skepsl** [NVlabs/sionna#533] [Q&A]
Hello, 

I have question about the properties of Paths, i.e., types. 
In the documentation, the shape of types is [batch_size, max_num_paths]. I am doing simple simulation with one Tx and two Rxs.

Here, other properties, e.g., mask, vertices, have batch : 2. However the types only has batch: 1.
This also happen no matter how many Rx or Tx I add into the simulation. While the other properties's batch follow accordingly, but the types's are batch still 1.
Does it mean that the types are all the same for entire batch,?

Thank you.

**Comment by jhoydis** [comment]
Hi @skepsl,

Can you share a simple working code example, ideally using one of the integrated scenes that reproduces what you observe?

**Reply by skepsl** [reply]
hello, I follow the tutorial in :
https://nvlabs.github.io/sionna/api/rt.html#scene
and adding one more rx.

So After executing scene.compute_path() then I inspect the properties, and found that types batch is only 1.

**Comment by skepsl** [comment]
So the code mainly constructed like these below:

```py
scene.tx_array = AntennaArray(antenna=Antenna(pattern='dipole',
                                                           polarization='V'),
                                           positions=np.array([[0.0, 0.0, 0.0]]))
scene.rx_array = AntennaArray(antenna=Antenna(pattern='iso',
                                                           polarization='V'),
                                           positions=np.array([[0.0, 0.0, 0.0]]))

#%%
tx = Transmitter(name='tx',
                 position=[0, 0, .5],
                 orientation=[.0, .0, .0],
                 )
scene.add(tx)
#%%
rx = Receiver(name='rx',
              position=[-26.3, -23.3, 0.17],
              orientation=[.0, .0, .0],
              )
scene.add(rx)
#%%
rx2 = Receiver(name='rx2',
              position=[-20., -23.3, 0.17],
              orientation=[.0, .0, .0],
              )
scene.add(rx2)
#%%
paths = scene.compute_paths(max_depth=4,
                            num_samples=1e4,
                            diffraction=True,
                            edge_diffraction=False,
                            scattering=False,
                            scat_keep_prob=0.001,
                            scat_random_phases=False,
                            check_scene=False)
```
and when I call the paths

```types = paths.types.numpy()
types
```

It only has batch: 1, here is [Batch, max_num_paths] : [1, 70] 
![image](https://github.com/user-attachments/assets/853eb76d-0602-4a11-85f5-6bcc26b6c18c)

while the others have batch: 2 
So does it mean that the types property is the same for every batch.?

Thanks

**Reply by faycalaa** [reply]
Hi @skepsl ,

The `batch_size` dimension for all properties of `Paths` is set to 1.

Regarding the code snippet you shared, could you let me know which scene you are using?
